Google.Ads.GoogleAds/examples/HotelAds/AddPerformanceMaxForTravelGoalsCampaign.cs

```diff
@@ -0,0 +1,1031 @@
+                request
+            );
+
+            Console.WriteLine($"Fetched a hotel asset suggestion for the place ID {placeId}");
+            return response.HotelAssetSuggestions[0];
+        }
+        // [END get_hotel_asset_suggestion]
+
+
+        ///<summary>
+        /// Creates multiple text assets and returns the list of resource names. The hotel asset
+        /// suggestion is used to create a text asset first. If the number of created text assets is
+        /// still fewer than the minimum required number of assets of the specified asset field
+        /// type, adds more text assets  to fulfill the requirement.
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID.</param>
+        /// <param name="assetFieldType">The asset field type that the text assets will be created
+        /// for.</param>
+        /// <param name="hotelAssetSuggestion">The hotel asset suggestion.</param>
+        /// <returns>The resource names of the created text assets.</returns>
+        /// </summary>
+        private List<string> CreateMultipleTextAssets(GoogleAdsClient client, long customerId,
+            AssetFieldType assetFieldType, HotelAssetSuggestion hotelAssetSuggestion)
+        {
+            // Uses the GoogleAdService to create multiple text assets in a single request.
+            List<MutateOperation> mutateOperations = new List<MutateOperation>();
+
+            // First, adds all the text assets of the specified asset field type.
+            // Filters to only the specified asset field type.
+            // Constructs a mutate operation to create the asset.
+            // Adds the operation to the list.
+            if (hotelAssetSuggestion.Status == HotelAssetSuggestionStatus.Success)
+            {
+                foreach (HotelTextAsset asset in hotelAssetSuggestion.TextAssets)
+                {
+                    if (asset.AssetFieldType == assetFieldType)
+                    {
+                        MutateOperation operation = new MutateOperation
+                        {
+                            AssetOperation = new AssetOperation
+                            {
+                                Create = new Asset
+                                {
+                                    TextAsset = new TextAsset
+                                    {
+                                        Text = asset.Text
+                                    }
+                                }
+                            }
+                        };
+                        mutateOperations.Add(operation);
+                    }
+                }
+            }
+
+            // If the added assets are still less than the minimum required assets for the asset
+            // field type, add more text assets using the default texts.
+            int i = 0;
+            while (mutateOperations.Count < MIN_REQUIRED_TEXT_ASSET_COUNTS[assetFieldType])
```

Comment:
Is `mutateOperations.Count` dynamically changed when you loop? If so, this might be a bug, since this will increase towards MIN_REQUIRED_TEXT_ASSET_COUNTS

Reply:
What do you mean? This adds mutateOperations until `mutateOperations` has `MIN_REQUIRED_TEXT_ASSET_COUNTS[assetFieldType]` elements, seems ok to me.

Reply:
Sorry. The bug will occur when we use the for loop. Since you use the while loop, it should be fine.